crates/weaver_live_check/src/sample_attribute.rs

```diff
@@ -0,0 +1,298 @@
+                            } else {
+                                // Mixed or unsupported types
+                                None
+                            }
+                        }
+                        None => None, // Empty or all-null array
+                    }
+                } else {
+                    None
+                }
+            }
+            Value::Object(_) => None, // Unsupported
+        }
+    }
+
+    fn update_stats(&mut self, stats: &mut LiveCheckStatistics) {
+        stats.inc_entity_count("attribute");
+        stats.maybe_add_live_check_result(self.live_check_result.as_ref());
+        let mut seen_attribute_name = self.name.clone();
+        if let Some(result) = &mut self.live_check_result {
+            for advice in &mut result.all_advice {
+                // If the advice is a template, adjust the name
+                if advice.advice_type == TEMPLATE_ATTRIBUTE_ADVICE_TYPE {
+                    if let Some(template_name) = advice.value.as_str() {
+                        seen_attribute_name = template_name.to_owned();
+                    }
+                }
+            }
+        }
+        stats.add_attribute_name_to_coverage(seen_attribute_name);
+    }
+}
+
+impl LiveCheckRunner for SampleAttribute {
+    fn run_live_check(&mut self, live_checker: &mut LiveChecker, stats: &mut LiveCheckStatistics) {
+        let mut result = LiveCheckResult::new();
+        // find the attribute in the registry
+        let semconv_attribute = {
+            if let Some(attribute) = live_checker.find_attribute(&self.name) {
+                Some(attribute.clone())
+            } else {
+                live_checker.find_template(&self.name).cloned()
+            }
+        };
+
+        if semconv_attribute.is_none() {
+            result.add_advice(Advice {
+                advice_type: MISSING_ATTRIBUTE_ADVICE_TYPE.to_owned(),
+                value: Value::String(self.name.clone()),
+                message: "Does not exist in the registry".to_owned(),
+                advice_level: AdviceLevel::Violation,
+            });
+        } else {
+            // Provide an info advice if the attribute is a template
+            if let Some(attribute) = &semconv_attribute {
+                if let AttributeType::Template(_) = attribute.r#type {
+                    result.add_advice(Advice {
+                        advice_type: TEMPLATE_ATTRIBUTE_ADVICE_TYPE.to_owned(),
+                        value: Value::String(attribute.name.clone()),
+                        message: "Is a template".to_owned(),
+                        advice_level: AdviceLevel::Information,
+                    });
+                }
+            }
+        }
+
+        // run advisors on the attribute
+        for advisor in live_checker.advisors.iter_mut() {
+            if let Ok(advice_list) = advisor.advise(
+                &SampleRef::Attribute(self),
```

Comment:
Here, you don't need to `&` the `SampleRef` if you pass the value, instead of reference.  You're effectively creating a new Ref for every call, but since `SampleRef` is so small/cheap this is fine.